File: rust/src/models/endpoint_headers_patch_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct EndpointHeadersPatchIn {
    pub headers: std::collections::BTreeMap<String, String>,

    /// A list of headers be be removed
    #[serde(rename = "deleteHeaders")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub delete_headers: Option<Vec<String>>,
}

impl EndpointHeadersPatchIn {
    pub fn new(headers: std::collections::BTreeMap<String, String>) -> Self {
        Self {
            headers,
            delete_headers: None,
        }
    }
}

File: javascript/src/models/endpointHeadersPatchIn.ts
// this file is @generated

export interface EndpointHeadersPatchIn {
  headers: { [key: string]: string };
  /** A list of headers be be removed */
  deleteHeaders?: string[];
}

export const EndpointHeadersPatchInSerializer = {
  _fromJsonObject(object: any): EndpointHeadersPatchIn {
    return {
      headers: object["headers"],
      deleteHeaders: object["deleteHeaders"],
    };
  },

  _toJsonObject(self: EndpointHeadersPatchIn): any {
    return {
      headers: self.headers,
      deleteHeaders: self.deleteHeaders,
    };
  },
};
